sql/moz-fx-data-shared-prod/firefox_desktop_derived/newtab_visits_daily_v2/query.sql

```diff
@@ -0,0 +1,284 @@
+  ANY_VALUE(newtab_search_enabled) AS newtab_search_enabled,
+  LOGICAL_OR(is_default_ui) AS is_default_ui,
+  LOGICAL_OR(event_category = 'newtab' AND event_name = 'opened') AS is_newtab_opened,
+  LOGICAL_OR(event_category = 'newtab.search' AND event_name = 'issued')
+  AND LOGICAL_OR(is_default_ui) AS is_search_issued,
+  LOGICAL_OR(event_category = 'newtab.search.ad' AND event_name = 'click')
+  AND LOGICAL_OR(is_default_ui) AS is_search_ad_click,
+  LOGICAL_OR(
+    event_category = 'pocket'
+    AND event_name IN ('click', 'dismiss', 'thumb_voting_interaction', 'thumb_voting_interaction')
+  )
+  AND LOGICAL_OR(is_default_ui) AS is_content_interaction,
+  LOGICAL_OR(event_category = 'pocket' AND event_name IN ('click'))
+  AND LOGICAL_OR(is_default_ui) AS is_content_click,
+  LOGICAL_OR(event_category = 'pocket' AND event_name IN ('impression'))
+  AND LOGICAL_OR(is_default_ui) AS is_content_impression,
+  LOGICAL_OR(
+    event_category = 'pocket'
+    AND event_name IN ('click', 'dismiss')
+    AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+  )
+  AND LOGICAL_OR(is_default_ui) AS is_sponsored_content_interaction,
+  LOGICAL_OR(
+    event_category = 'pocket'
+    AND event_name IN ('click')
+    AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+  )
+  AND LOGICAL_OR(is_default_ui) AS is_sponsored_content_click,
+  LOGICAL_OR(
+    event_category = 'pocket'
+    AND event_name IN ('impression')
+    AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+  )
+  AND LOGICAL_OR(is_default_ui) AS is_sponsored_content_impression,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(event_category = 'pocket' AND event_name IN ('click')),
+    0
+  ) AS any_content_click_count,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(event_category = 'pocket' AND event_name IN ('impression')),
+    0
+  ) AS any_content_impression_count,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(
+      event_category = 'pocket'
+      AND event_name IN ('click')
+      AND NOT SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ),
+    0
+  ) AS organic_content_click_count,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(
+      event_category = 'pocket'
+      AND event_name IN ('impression')
+      AND NOT SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ),
+    0
+  ) AS organic_content_impression_count,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(
+      event_category = 'pocket'
+      AND event_name IN ('click')
+      AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ),
+    0
+  ) AS sponsored_content_click_count,
+  IF(
+    LOGICAL_OR(is_default_ui),
+    COUNTIF(
+      event_category = 'pocket'
+      AND event_name IN ('impression')
+      AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
+    ),
+    0
+  ) AS sponsored_content_impression_count,
```

Comment:
Why are we using both `IF` and `LOGICAL_OR` here?  It seems redundant.  As far as I can tell replacing the following could replace lines 163-171 and do the same thing with fewer operations:

```
 COUNTIF(
      is_default_ui
      AND event_category = 'pocket'
      AND event_name IN ('impression')
      AND SAFE_CAST(mozfun.map.get_key(event_details, 'is_sponsored') AS BOOLEAN)
    ) AS sponsored_content_impression_count
  ```

Reply:
For the same reasons i answered above...they (is_default_ui and event_category = 'pocket') are different events. We need to apply the condition to the result of the aggregations

Reply:
Okay I am seeing how they are different now.  But in the case that there were records with the same visit_id where `is_default_ui` had different values, in the current setup we'd count ALL interactions whether or not `is_default_ui` was true or not.  It might be better to count only the "valid" interactions, the ones where the interactions were associated with `is_default_ui` being true.

Reply:
> But in the case that there were records with the same visit_id where is_default_ui had different values, in the current setup we'd count ALL interactions whether or not is_default_ui was true or not

The way we are implementing this is, if a visit was a default ui, then all other activity associated with the visit is attributed to default ui. 
Note: the way we associate other activity to a corresponding newtab open (or a default ui) is based on the visit id.
Do you have any analysis to illustrate, with relevant data, to associate specific interactions within the same visit to default ui and some others that are specific to non-default, could you share those analysis with all of us?

I am going to mark this as resolved because this implementation is as expected

Reply:
Please stop marking these as resolved, I need to understand what is happening here before I sign off on it and I'm still not there.

Because we're aggregating over is_default_ui, it's possible we could get a single visit with data like:

is_default_ui: True, event_name: impression
is_default_ui: True, event_name: click
is_default_ui: False, event_name: impression
is_default_ui: False, event_name: click

In the current logic, we'd count two clicks which doesn't seem right.

Reply:
Working on figuring this out

Reply:
Also please leave a comment here about what's going on with the different events and how that affects the aggregation, I clearly got tripped up on it 😅

Reply:
> This only works if we can have it be NULL rather than False for events that aren't a tab open though

If you are referring to the udf is_default_ui that returns a _boolean_ `true` when a default ui condition is met, then that is how its expected to work. The function is un-opinionated on situations that are _not_ "default ui"

> Also please leave a comment here about what's going on with the different events and how that affects the aggregation, I clearly got tripped up on it 😅

I will add a comment explaining how the aggregation with the `LOGICAL_OR(NOT is_default_ui)` is expected to work. Thanks for the suggestion!

Reply:
I have added inline comment for the usage of LOGICAL_OR(NOT is_default_ui)